### ISSUE TITLE
Problem with unique constraints dependencies?

### ISSUE THREAD

**Issue by dvarrazzo** [perseas/Pyrseas#169]
I have a problem with [this piece of code](https://github.com/perseas/Pyrseas/blob/cd3f4e9/pyrseas/dbobject/constraint.py#L353-L356), the current master:

```python
    def _find_referenced_index(self, db, ref_table):
        # ...
        if hasattr(ref_table, 'unique_constraints'):
            for uc in list(ref_table.unique_constraints.values()):
                if uc.keycols == self.ref_cols:
                    return uc
        # ...
```

this crashes just because ref_table.unique_constraints contains dictionaries, not objects, so `uc.keycols` cannot be used. The dictionary seems like:

```
'unique_constraints': {u'CONSTRAINT_NAME': {'columns': [u'FIELD_NAME']}
```

if the program passed this function it would crash downstream as the dict `uc` is passed to `get_implied_deps()` which would crash on `deps.add(idx)` as the dict is not hashable. It seems this function would like to return a pyrseas object, but I don't know what.

I wonder why are tests passing on master and crashing on the db I'm trying to dump? This code seems wrong...

**Comment by jmafc** [comment]
The merging of various changes dealing with `keycols` that had different names in `r0.7` and `master` were a pain in the rear, so I'm not surprised. I'll check, probably Monday. I assume your db has unique constraints with multiple columns and that something like that should reproduce it?  Do you have a traceback?

**Comment by jmafc** [comment]
@dvarrazzo Could you please check this against the latest master?

**Comment by jmafc** [comment]
@dvarrazzo I believe that whatever problems you may have had with the code shown in your initial comment have been addressed in part by changes made to release 0.8 and further by change cfc7f95, i.e., using a deep copy for DbObject. So I'm closing this. If you find otherwise, please reopen.
